akka-samples/akka-docs-java-lambda/src/test/java/docs/stream/RateTransformationDocTest.java

```diff
@@ -0,0 +1,164 @@
+    //#conflate-summarize
+    final Flow<Double, Tuple3<Double, Double, Integer>, BoxedUnit> statsFlow =
+      Flow.of(Double.class)
+        .conflate(elem -> {
+          return new ArrayList<Double>() {
+            {
+              this.add(elem);
+            }
+          };
+        } , (acc, elem) -> {
+          acc.add(elem);
+          return acc;
+        })
+        .map(s -> {
+          final Double mean = s.stream().mapToDouble(d -> d).sum() / s.size();
+          final DoubleStream se = s.stream().mapToDouble(x -> Math.pow(x - mean, 2));
+          final Double stdDev = Math.sqrt(se.sum() / s.size());
+          return new Tuple3(stdDev, mean, s.size());
+        });
+    //#conflate-summarize
+
+    final Future<List<Tuple3<Double, Double, Integer>>> fut = Source.repeat(0).map(i -> r.nextGaussian())
+      .via(statsFlow)
+      .grouped(10)
+      .runWith(Sink.head(), mat);
+
+    final Duration timeout = Duration.create(100, TimeUnit.MILLISECONDS);
+    Await.result(fut, timeout);
+  }
+
+  @Test
+  public void conflateShouldSample() throws Exception {
+    //#conflate-sample
+    final Double p = 0.01;
+    final Flow<Double, Double, BoxedUnit> sampleFlow = Flow.of(Double.class)
+      .conflate(elem -> {
+        return new ArrayList<Double>() {
+          {
+            this.add(elem);
+          }
+        };
+      } , (acc, elem) -> {
+        if (r.nextDouble() < p) {
+          acc.add(elem);
+        }
+        return acc;
+      })
+      .mapConcat(d -> d);
+    //#conflate-sample
+
+    final Future<Double> fut = Source.from(new ArrayList<Double>(Collections.nCopies(10000, 1.0)))
+      .via(sampleFlow)
+      .runWith(Sink.fold(0.0, (agg, next) -> agg + next), mat);
+
+    final Duration timeout = Duration.create(1, TimeUnit.SECONDS);
+    final Double count = Await.result(fut, timeout);
+  }
+
+  @Test
+  public void expandShouldRepeatLast() throws Exception {
+    //#expand-last
+    final Flow<Double, Double, BoxedUnit> lastFlow = Flow.of(Double.class)
+      .expand(d -> d, s -> new Pair(s, s));
+    //#expand-last
+
+    final Pair<TestPublisher.Probe<Double>, Future<List<Double>>> probeFut = TestSource.<Double> probe(system)
+      .via(lastFlow)
+      .grouped(10)
+      .toMat(Sink.head(), Keep.both())
+      .run(mat);
+
+    final TestPublisher.Probe<Double> probe = probeFut.first();
+    final Future<List<Double>> fut = probeFut.second();
+    probe.sendNext(1.0);
+    final Duration timeout = Duration.create(1, TimeUnit.SECONDS);
+    final List<Double> expanded = Await.result(fut, timeout);
+    assertEquals(expanded.size(), 10);
+    assertEquals(expanded.stream().mapToDouble(d -> d).sum(), 10, 0.1);
+  }
+
+  @Test
+  public void expandShouldTrackDrift() throws Exception {
+    //#expand-drift
+    final Flow<Double, Pair<Double, Integer>, BoxedUnit> driftFlow = Flow.of(Double.class)
+      .expand(d -> new Pair<Double, Integer>(d, 0), t -> {
```

Comment:
Yup. Without `<Double, Integer>` id does not know that `t.second()` is and `Integer` in the second parameter lambda.